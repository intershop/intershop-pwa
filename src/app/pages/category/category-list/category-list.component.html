<ul class="category-list row">
  <li *ngFor="let category of categories" class="category-list-item col-6 col-lg-4">
    <ish-category-tile [categoryUniqueId]="category"></ish-category-tile>
  </li>
</ul>
