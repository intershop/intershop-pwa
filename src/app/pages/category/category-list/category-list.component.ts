import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'ish-category-list',
  templateUrl: './category-list.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CategoryListComponent {
  @Input() categories: string[];
}
