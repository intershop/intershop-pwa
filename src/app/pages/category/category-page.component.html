<ng-container *ngIf="category$ | async as category">
  <!-- category with subcategories -->
  <ng-container *ngIf="category.hasChildren; else productListing">
    <ish-category-categories [category]="category" [deviceType]="deviceType$ | async"></ish-category-categories>
  </ng-container>

  <!-- category with products -->
  <ng-template #productListing>
    <ng-container *ngIf="category.hasOnlineProducts">
      <ish-category-products [category]="category" [deviceType]="deviceType$ | async"></ish-category-products>
    </ng-container>
  </ng-template>

  <!-- empty category without subcategories and without products -->
  <ng-container *ngIf="!category.hasChildren && !category.hasOnlineProducts">
    <!-- TODO: decide how to handle empty categories -->
    <h1>{{ category.name }}</h1>
  </ng-container>
</ng-container>
