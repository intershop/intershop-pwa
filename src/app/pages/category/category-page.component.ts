import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs';

import { AppFacade } from 'ish-core/facades/app.facade';
import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
import { CategoryView } from 'ish-core/models/category-view/category-view.model';
import { DeviceType } from 'ish-core/models/viewtype/viewtype.types';

@Component({
  selector: 'ish-category-page',
  templateUrl: './category-page.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CategoryPageComponent implements OnInit {
  category$: Observable<CategoryView>;
  deviceType$: Observable<DeviceType>;

  constructor(private shoppingFacade: ShoppingFacade, private appFacade: AppFacade) {}

  ngOnInit() {
    this.category$ = this.shoppingFacade.selectedCategory$;
    this.deviceType$ = this.appFacade.deviceType$;
  }
}
