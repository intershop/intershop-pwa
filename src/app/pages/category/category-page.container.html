<ng-container *ngIf="category$ | async as category">
  <!-- category with subcategories -->
  <ng-container *ngIf="category.hasChildren(); else productListing">
    <ish-category-page [category]="category" [deviceType]="deviceType$ | async"></ish-category-page>
  </ng-container>

  <!-- category with products -->
  <ng-template #productListing>
    <ng-container *ngIf="category.hasOnlineProducts">
      <ish-family-page [category]="category" [deviceType]="deviceType$ | async"></ish-family-page>
    </ng-container>
  </ng-template>

  <!-- empty category without subcategories and without products -->
  <ng-container *ngIf="!category.hasChildren() && !category.hasOnlineProducts">
    <!-- TODO: decide how to handle empty categories -->
    <h1>{{ category.name }}</h1>
  </ng-container>
</ng-container>

<ish-loading *ngIf="categoryLoading$ | async"></ish-loading>
