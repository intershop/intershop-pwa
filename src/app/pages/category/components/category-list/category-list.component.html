<ul class="category-list row">
  <li *ngFor="let category of categories; trackBy: trackByFn" class="category-list-item col-6 col-lg-4">
    <ish-category-tile [category]="category"></ish-category-tile>
  </li>
</ul>
