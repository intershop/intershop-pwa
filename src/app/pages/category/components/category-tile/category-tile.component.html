<div class="category-name" [attr.data-testing-id]="'category-' + category.uniqueId">
  <a [routerLink]="['/category', category.uniqueId]">
    <ish-category-image [category]="category"></ish-category-image>
    <h3>{{ category.name }}</h3>
  </a>
</div>
