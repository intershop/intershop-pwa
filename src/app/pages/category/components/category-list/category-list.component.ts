import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { CategoryView } from 'ish-core/models/category-view/category-view.model';
import { Category } from 'ish-core/models/category/category.model';

@Component({
  selector: 'ish-category-list',
  templateUrl: './category-list.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CategoryListComponent {
  @Input() categories: Category[];

  trackByFn(_, item: CategoryView) {
    return item.uniqueId;
  }
}
