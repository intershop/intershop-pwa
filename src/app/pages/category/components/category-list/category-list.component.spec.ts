import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { MockComponent } from 'ng-mocks';

import { Category } from 'ish-core/models/category/category.model';

import { CategoryTileComponent } from '../category-tile/category-tile.component';

import { CategoryListComponent } from './category-list.component';

describe('Category List Component', () => {
  let component: CategoryListComponent;
  let fixture: ComponentFixture<CategoryListComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [CategoryListComponent, MockComponent(CategoryTileComponent)],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CategoryListComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    component.categories = [
      { uniqueId: 'uid1', name: 'name1', images: [{ effectiveUrl: '/url1.png' }] },
      { uniqueId: 'uid2', name: 'name2', images: [{ effectiveUrl: '/url2.png' }] },
    ] as Category[];
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchInlineSnapshot(`
      <ul class="category-list row">
        <li class="category-list-item col-6 col-lg-4"><ish-category-tile></ish-category-tile></li>
        <li class="category-list-item col-6 col-lg-4"><ish-category-tile></ish-category-tile></li>
      </ul>
    `);
  });
});
