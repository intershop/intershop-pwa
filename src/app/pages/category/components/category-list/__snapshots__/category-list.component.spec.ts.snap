// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category List Component should be created 1`] = `
<div
  id="root0"
  ng-version="7.1.2"
>
  <ul
    class="category-list row"
  >
    
    <li
      class="category-list-item col-6 col-lg-4"
    >
      <div
        class="category-name"
        data-testing-id="category-uid1"
      >
        <a
          href="/category/uid1"
          ng-reflect-router-link="/category,uid1"
        >
          
          <img
            data-testing-id="category-image-"
            height="270"
            src="http://www.example.org/url1.png"
            width="270"
          />
          <h3>
            name1
          </h3>
        </a>
      </div>
    </li>
    <li
      class="category-list-item col-6 col-lg-4"
    >
      <div
        class="category-name"
        data-testing-id="category-uid2"
      >
        <a
          href="/category/uid2"
          ng-reflect-router-link="/category,uid2"
        >
          
          <img
            data-testing-id="category-image-"
            height="270"
            src="http://www.example.org/url2.png"
            width="270"
          />
          <h3>
            name2
          </h3>
        </a>
      </div>
    </li>
  </ul>
</div>
`;
